File: Beecrowd/1 - Beginner/2003 - 2176/2126 - Searching Subsequences/Python/2126_SearchingSubsequences.py
count = 0
pos = 0
i = 1

while True:
    try:
        subsequence = input()
        sequence = input()

        while pos != -1:
            pos = sequence.find(subsequence, pos)

            if pos != -1:
                count += 1
                pos += 1

        print(f"Caso #{i}:")
        if count == 0:
            print("Nao existe subsequencia")
        else:
            print(f"Qtd.Subsequencias: {count}")
            print(f"Pos: {sequence.rfind(subsequence) + 1}")

        print()
        
        count = pos = 0
        
        i += 1

    except EOFError:
        break

File: Beecrowd/1 - Beginner/2003 - 2176/2126 - Searching Subsequences/2126_SearchingSubsequences.c
#include <stdio.h>
#include <string.h>

int main() {
    char sequence[100], subsequence[100];
    int count = 0, pos = 0, i = 1;

    while (scanf("%s%s", subsequence, sequence) != EOF) {
        char *ptr = sequence;

        while ((ptr = strstr(ptr, subsequence)) != NULL) {
            count++;
            pos = (int)(ptr - sequence) + 1;
            ptr++;
        }

        printf("Caso #%d:\n", i);

        if (count == 0) {
            printf("Nao existe subsequencia\n");
        } else {
            printf("Qtd.Subsequencias: %d\n", count);
            printf("Pos: %d\n", pos);
        }
        
        printf("\n");

        count = 0, pos = 0;

        i++;
    }

    return 0;
}